### ISSUE TITLE
Duplicate Collection if postmanUid fails IsGuid check

### ISSUE THREAD

**Issue by jbrinkman** [apideck-libraries/portman#88]
If a user mistypes the postmanUid when executing Portman, it will create a new collection, even if one already exists with the same name.  This bug occurs in `syncCollectionToPostman()` of `Portman.ts` because the calculation of the collection name occurs before the check for a valid postman UID.

```typescript
      const collectionIdentification = postmanUid || (portmanCollection?.info?.name as string)
      const postman = new PostmanService()
      if (postman.isGuid(collectionIdentification)) {
        await postman.updateCollection(portmanCollection, collectionIdentification)
      } else {
        const remoteCollection = (await postman.findCollectionByName(
          collectionIdentification
        )) as Record<string, unknown>
```

With this logic a failure of `postman.isGuid` will also result in a failure for `postman.findCollectionByName` since the collectionidentification was set using the postmanUid option which will never match a Collection Name.

**Comment by thim81** [comment]
The postman sync logic can be improved and optimized (#41)

We will see what we can do to fix this. 

If it blocks you, while waiting for the fix, you can skip the uid setting. The collection will be updated instead of a newly created next to it, based on the name.

**Comment by jbrinkman** [comment]
This is not a blocker.  We'll just make sure to put in a valid Guid ;)

**Comment by thim81** [comment]
@jbrinkman status update: a PR is in the making, that will include a fix for this item. It will be part of the optimisation of the Postman sync flow.

**Comment by thim81** [comment]
@jbrinkman The issue should be resolved as part of 1.4.2 release. We included a error message when an incorrect Postman UID is being used, instead of creating a new collection.
